project/Boilerplate.scala

```diff
@@ -317,8 +322,8 @@ object Boilerplate {
         -  /**
         -   * @group Product
         -   */
-        -  final def forProduct$arity[Source, ${`A..N`}]($memberNames)(f: Source => $outputType)(implicit
-        -    $instances, dummy: DummyImplicit
+        -  final def forTupleProduct$arity[Source, ${`A..N`}]($memberNames)(f: Source => $outputType)(implicit
```

Comment:
Bikeshedding of new name is needed.
forNamedProduct, ofProduct are some suggestions.

Reply:
I agree, a new name is needed.   Could we just use `fromProductTyped$arity`?   Also, I feel like we should only generate these for Scala3 to avoid any confusion or issues in Scala2.

Reply:
if we are only generating this for scala 3, then we could keep the original name and just change the type to tuple.

Reply:
I would rather keep the new name, I think my previous concerns with the change of the signature still apply even if this change is only for scala 3.